Bottom: ecee8d15639b1da818402f3ffd74ced8784344df
Top:    2721bf85ce00aa1b4bd0984155493635007abc66
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-21 21:55:20 +0100

Add debug catch to triangle handling

If for any reason we are about to run off the end of the
triangle array, deliberately segfault so gdb can catch where.


---

diff --git a/src/hid/common/hidgl.h b/src/hid/common/hidgl.h
index aa2985e..a3e966f 100644
--- a/src/hid/common/hidgl.h
+++ b/src/hid/common/hidgl.h
@@ -43,6 +43,13 @@ hidgl_add_triangle_3D (triangle_buffer *buffer,
                        GLfloat x2, GLfloat y2, GLfloat z2,
                        GLfloat x3, GLfloat y3, GLfloat z3)
 {
+  if (buffer->triangle_count == TRIANGLE_ARRAY_SIZE)
+    {
+      fprintf (stderr, "hidgl_add_triangle_3D: "
+               "Attempt to add past end of array. Will now segfault.\n");
+      *(char *)0 = 0;
+    }
+
   buffer->triangle_array [buffer->coord_comp_count++] = x1;
   buffer->triangle_array [buffer->coord_comp_count++] = y1;
   buffer->triangle_array [buffer->coord_comp_count++] = z1;
